ui/src/components/CreateClusterForm.js

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui/styles';
+import { InputLabel } from 'material-ui/Input';
+import TextField from 'material-ui/TextField';
+import Typography from 'material-ui/Typography';
+import { MenuItem } from 'material-ui/Menu';
+import { FormControl } from 'material-ui/Form';
+import Select from 'material-ui/Select';
+import Button from 'material-ui/Button';
+import { CircularProgress } from 'material-ui/Progress';
+
+import { createApiUrl } from '../net'
+
+
+const DEFAULT_MASTER_MACHINE_TYPE = "n1-standard-4";
+const DEFAULT_WORKER_MACHINE_TYPE = "n1-standard-2";
+const DEFAULT_WORKER_DISK_SIZE = 200;
+
+// Regex for GCP Project names.
+const projectRegex = /^[a-z][a-z0-9_-]{0,50}[a-z0-9]$/;
+
+// Regex for cluster names.
+const clusterRegex = /^[a-z][a-z0-9-]{0,25}[a-z0-9]$/;
+
+const validMachineTypes = [
```

Comment:
Why limit it to this set? What if someone needs a highmem machine type?

Reply:
Added a few more types